src/main/scala/tailRecursion/TailRecursion.scala

```diff
@@ -3,27 +3,21 @@ package tailRecursion
 import java.util.Scanner
 
 object TailRecursion {
-  def main(args: Array[String]): Unit = {
-    val sc: Scanner = new Scanner(System.in)
-    println(recursionIntToString(sc.nextInt()))
-  }
-
-
   def recursionIntToString(n: Int): String ={
     import scala.annotation.tailrec
     @tailrec // will fail compilation if method can’t be optimized
-    def recursionIntToStringInternal (result: String, tempN: Int): String = {
+    def recursionIntToStringInternal (result: Array[Int], tempN: Int): String = {
       if (tempN <= 0) {
-        s"$result${reverseString(s"$result$tempN,")}"
+        resultString(result, tempN)
       } else {
-        recursionIntToStringInternal(s"$result$tempN,", tempN - 4)
+        recursionIntToStringInternal(result :+ tempN, tempN - 4)
```

Comment:
Create a constant or parameter for "4" literal. Please, don`t use "magic numbers".

Reply:
step variable has been added to TailRecursion class. This variable is used in recursion function instead of literal